src/common.cpp

```diff
@@ -76,16 +74,32 @@ std::vector<shape::dynamic_dimension> compute_broadcasted_dyn_dims(shape s0, sha
                        {
                            return b;
                        }
+                       else if(a.within_range(b))
+                       {
+                           return shape::dynamic_dimension{a.min, a.max};
```

Comment:
why not return just 'a' ?

Reply:
don't want to include the optimals value if it exists